packages/react/src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,93 @@
+import { Slot } from '@radix-ui/react-slot';
+import cl from 'clsx/lite';
+import {
+  Fragment,
+  type HTMLAttributes,
+  type ReactNode,
+  forwardRef,
+  useMemo,
+} from 'react';
+
+export type AvatarProps = {
+  /**
+   * The name of the person the avatar represents.
+   * Will be used to generate initials if no image is provided.
+   */
+  name?: string;
+  /**
+   * The color of the avatar.
+   *
+   * @default 'accent'
+   */
+  color?: 'accent' | 'neutral' | 'brand1' | 'brand2' | 'brand3';
+  /**
+   * The size of the avatar.
+   *
+   * @default 'md'
+   */
+  size?: 'xs' | 'sm' | 'md' | 'lg';
+  /**
+   * The shape of the avatar.
+   *
+   * @default 'circle'
+   */
+  variant?: 'circle' | 'square';
+  /**
+   * Image or icon to display inside the avatar.
+   *
+   * Gets `aria-hidden="true"`
+   */
+  children?: ReactNode;
+} & HTMLAttributes<HTMLSpanElement>;
+
+const fontSizeMap = {
+  xs: 'ds-paragraph--xs',
+  sm: 'ds-heading--2xs',
+  md: 'ds-heading--sm',
+  lg: 'ds-heading--md',
+};
+
+export const Avatar = forwardRef<HTMLSpanElement, AvatarProps>(function Avatar(
+  {
+    name,
+    color = 'accent',
+    size = 'md',
+    variant = 'circle',
+    className,
+    children,
+    ...rest
+  },
+  ref,
+) {
+  const Component = children ? Slot : Fragment;
+
+  const initials = useMemo(() => {
+    return getInitials(name);
+  }, [name]);
+
+  return (
+    <span
+      ref={ref}
+      className={cl('ds-avatar', fontSizeMap[size], className)}
+      data-ds-variant={variant}
+      data-ds-color={color}
+      data-ds-size={size}
+      {...rest}
+      role='img'
+      aria-label={name}
+    >
+      <Component {...(children ? { 'aria-hidden': true } : {})}>
+        {children || initials}
+      </Component>
+    </span>
+  );
+});
+
+/**
+ * Gets initials using first and last word of a name.
+ */
+function getInitials(name: string | undefined): string | null {
+  if (!name) return null;
+  const splitName = name.trim().split(' ');
```

Comment:
If we want to support serious internationalization, we could instead do:
```
function getInitials(name: string | undefined): string | null {
  if (!name) return null;
  const initialis = [];
  const segments = new Intl.Segmenter(document.documentElement.lang, { granularity: 'word' }).segment(name)
  for (const segment of segments) if (segment.isWordLike) initialis.push(segment.segment);
  return initialis.slice(0, 2).join('').toUpperCase();
}
```

Reply:
Could you elaborate? We only want the first letter of the first and last word

Reply:
In Norwegian yes, but some languages i.e. consider `-` a word break, or have characters that should be skipped to get the actual first letter. Using `Intl.Segmenter` ensures we have international support right out of the box, see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Segmenter#examples

Reply:
Right, good catch. The code you sent is not quite right, but I'll tweak it! 🥇